counsel.el

```diff
@@ -379,9 +379,13 @@ Update the minibuffer with the amount of lines collected every
     (when len
       (setq ivy-completion-beg (- (point) len))
       (setq ivy-completion-end (point))
-      (ivy-read "Candidate: " company-candidates
-                :action #'ivy-completion-in-region-action
-                :caller 'counsel-company))))
+      (cond
+       ((= (length company-candidates) 1)
+        (ivy-completion-in-region-action (car company-candidates))
+        )
+       (t (ivy-read "Candidate: " company-candidates
+                  :action #'ivy-completion-in-region-action
+                  :caller 'counsel-company))))))
```

Comment:
Nit: there's a trailing paren and the indentation seems off. Suggest to simplify:

```el
(if (cdr company-candidates)
    (ivy-read "Candidate: " company-candidates
              :action #'ivy-completion-in-region-action
              :caller 'counsel-company)
  (ivy-completion-in-region-action (car company-candidates)))
```